apps/server/src/shared/domain/entity/task.entity.ts

```diff
@@ -99,102 +104,153 @@ export class Task extends BaseEntityWithTimestamps implements ILearnroomElement,
 		this.publicSubmissions = props.publicSubmissions || false;
 	}
 
-	isFinishedForUser(user: User): boolean {
-		return !!(this.finished?.contains(user) || this.course?.isFinished());
+	private getSubmissionItems(): Submission[] {
+		if (!this.submissions || !this.submissions.isInitialized(true)) {
+			throw new InternalServerErrorException('Submissions items are not loaded.');
+		}
+		const submissions = this.submissions.getItems();
+
+		return submissions;
+	}
+
+	private getFinishedUserIds(): EntityId[] {
+		const finishedObjectIds = this.finished.getIdentifiers('_id');
+		const finishedIds = finishedObjectIds.map((id): string => id.toString());
+
+		return finishedIds;
 	}
 
-	isDraft(): boolean {
+	private getParent(): ITaskParent | User {
+		const parent = this.lesson || this.course || this.creator;
+
+		return parent;
+	}
+
+	private getMaxSubmissions(): number {
+		const parent = this.getParent();
+		// For draft (user as parent) propaly user is not a student, but for maxSubmission one is valid result
+		const maxSubmissions = parent instanceof User ? 1 : parent.getStudentIds().length;
+
+		return maxSubmissions;
+	}
+
+	private isFinishedForUser(user: User): boolean {
+		const finishedUserIds = this.getFinishedUserIds();
+		const isUserInFinishedUser = finishedUserIds.some((finishedUserId) => finishedUserId === user.id);
+
+		const isCourseFinished = this.course ? this.course.isFinished() : false;
+
+		const isFinishedForUser = isUserInFinishedUser || isCourseFinished;
+
+		return isFinishedForUser;
+	}
+
+	public isDraft(): boolean {
 		// private can be undefined in the database
 		return !!this.private;
 	}
 
-	isPublished(): boolean {
+	public isPublished(): boolean {
 		if (this.isDraft()) {
 			return false;
 		}
 		if (this.availableDate && this.availableDate > new Date(Date.now())) {
 			return false;
 		}
+
 		return true;
 	}
 
-	isPlanned(): boolean {
+	public isPlanned(): boolean {
 		if (this.isDraft()) {
 			return false;
 		}
 		if (this.availableDate && this.availableDate > new Date(Date.now())) {
 			return true;
 		}
+
 		return false;
 	}
 
-	areSubmissionsPublic(): boolean {
-		return !!this.publicSubmissions;
+	private getSubmittedSubmissions(): Submission[] {
+		const submissions = this.getSubmissionItems();
+		const gradedSubmissions = submissions.filter((submission) => submission.isSubmitted());
```

Comment:
Name of the const should be "submittedSubmissions".

Reply:
true